include/revng/ADT/GenericGraph.h

```diff
@@ -230,6 +246,28 @@ class ForwardNode
   bool hasSuccessors() const { return Successors.size() != 0; }
   size_t successorCount() const { return Successors.size(); }
 
+  template<
+    typename = std::enable_if_t<detail::CanBeIndexed<NeighborContainer, Edge>>>
```

Comment:
This is a bit overkill. `enable_if` is quite hard to read and, thanks to concepts, we've been able to get rid of them all.
Also this always evaluates to true, since the `NeighborContainer` is not user-customziable (it's always a `SmallVector`).

Reply:
Yes, I know about `SmallVector` being backed in, it's more of a defense against the future changes that would potentially break this. Either way, dropping SFINAE is definitely no big deal, my concerns were more about introducing random access to begin with.
The only reason for only using such non-committal ways of accessing the members I can think of is to keep the underlying data-structure flexible in case we want to eventually change it.

But, anyway, so you have nothing against introducing random access here, right?

Reply:
OKGO for random access.
If you really want to verify this, put a `static_assert` after the definition of `NeighborContainer`.
If we change the container the `static_assert` will fail and we can either fix or drop.

Reply:
How do you like current iteration?

Reply:
Looks better, but it's still a bit redundant to put a `static_assert` each time, given that the property you're checking is `class`-wise, not method-wise.
It'd make sense to have it multiple times if there was a template parameter in the method.
I'd go for just a single assertion.

Reply:
Then there's a doubt as to where should I put it. At this point, maybe it's better to drop it entirely: if we change `SmallVector` to something with no `operator[](size_t)` support, it'll cause a compilation problems either way. The purpose of the assertion was more to say "if you change the vector to something else, you need to fix this function here", but if we put it somewhere unrelated, there's no real reason to include the assert.

So, I'm just dropping them then?

Reply:
Done.